src/commands/react-native/codepush.ts

```diff
@@ -0,0 +1,184 @@
+import {exec} from 'child_process'
+import {Cli, Command} from 'clipanion'
+import {CodepushHistoryCommandError, CodepushHistoryParseError, NoCodepushReleaseError} from './errors'
+
+import {RNPlatform, RN_SUPPORTED_PLATFORMS} from './interfaces'
+import {UploadCommand} from './upload'
+
+export class CodepushCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Upload your React Native Codepush bundle and sourcemaps to Datadog.',
+    details: `
+    This command will upload React Native Codepush sourcemaps and their corresponding javascript bundle to Datadog in order to un-minify front-end stack traces received by Datadog.
+    See README for details.
+    `,
+    examples: [
+      [
+        'Upload ios staging sourcemaps for Company/AppNameiOS',
+        'datadog-ci react-native codepush --platform ios --service com.company.app --bundle ./build/main.jsbundle --sourcemap ./build/main.jsbundle.map --app Company/AppNameiOS --deployment Staging',
+      ],
+      [
+        'Upload android production sourcemaps for Company/AppNameAndroid',
+        'datadog-ci react-native codepush --platform android --service com.company.app --bundle ./build/index.android.bundle --sourcemap ./build/index.android.bundle.map --app Company/AppNameAndroid --deployment Production',
+      ],
+    ],
+  })
+
+  private appCenterAppName?: string
+  private appCenterDeployment?: string
+  /**
+   * There should not be multiple uploads with the same version in the case
+   * of codepush, so we can go with a default of "1".
+   */
+  private buildVersion = '1'
+  private bundle?: string
+  private configPath?: string
+  private disableGit?: boolean
+  private dryRun = false
+  private maxConcurrency = 20
+  private platform?: RNPlatform
+  private projectPath?: string
+  private releaseVersion?: string
+  private repositoryURL?: string
+  private service?: string
+  private sourcemap?: string
+
+  public async execute() {
+    if (!this.service) {
+      this.context.stderr.write('Missing service\n')
+
+      return 1
+    }
+
+    if (!this.bundle) {
+      this.context.stderr.write('Missing bundle path\n')
+
+      return 1
+    }
+
+    if (!this.platform) {
+      this.context.stderr.write('Missing platform\n')
+
+      return 1
+    }
+
+    if (!RN_SUPPORTED_PLATFORMS.includes(this.platform)) {
+      this.context.stderr.write(
+        `Platform ${this.platform} is not supported.\nSupported platforms are ios and android.\n`
+      )
+
+      return 1
+    }
+
+    if (!this.sourcemap) {
+      this.context.stderr.write('Missing sourcemap file path\n')
+
+      return 1
+    }
+
+    if (!this.appCenterAppName) {
+      this.context.stderr.write('Missing AppCenter app name\n')
+
+      return 1
+    }
+
+    if (!this.appCenterDeployment) {
+      this.context.stderr.write('Missing AppCenter deployment\n')
+
+      return 1
+    }
```

Comment:
One thing that was suggested for the `flutter-symbols upload` command was to try to try to group a lot of this verification so that you get all your missing parameters at once, instead of running, seeing an error, fixing it, and then running again.

Reply:
That's a good point, I'll probably make this change for all react-native commands in a separate PR to improve them as it is not a small change